koiki/create_delivery.py

```diff
@@ -33,6 +33,12 @@ def _deliveries(self):
 
         return deliveries
 
+    def _vendors(self):
```

Comment:
I think it's simpler and more performant to iterate over line items and get the vendors out of them, removing duplicates. Also, I would implement it as a method on `Order` so that we can call `order['vendors']` from the `Client` itself, without having to call this class.

Similarly, I would move `by_vendor` also to `Order` so all the order-related data are encapsulated together.

In the end, this class aimed to encapsulate the logic related to the endpoint although its responsibilities are becoming a bit blurry.

Reply:
I've refactored moving the logic away from CreateDelivery and Client classes into Order and Delivery classes so that it looks better encapsulated and I profit the line_items iteration to obtain the self.vendors array to be used later on.